lms/djangoapps/student_account/views.py

```diff
@@ -302,6 +306,69 @@ def _external_auth_intercept(request, mode):
         return external_auth_register(request)
 
 
+def get_user_orders(user):
+    """Given a user, get the detail of all the orders from the Ecommerce service.
+
+    Arguments:
+        user (User): The user to authenticate as when requesting ecommerce.
+
+    Returns:
+        list of dict, representing orders returned by the Ecommerce service.
+    """
+    no_data = []
+    commerce_configuration = CommerceConfiguration.current()
+    if not commerce_configuration.enabled:
+        log.warning('%s configuration is disabled.', commerce_configuration.API_NAME)
+        return no_data
+
+    if commerce_configuration.is_cache_enabled:
+        cache_key = commerce_configuration.CACHE_KEY + '.' + user.username
```

Comment:
Please use the user ID to cache this information. The username is public and goes through a hash function [here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/memcache.py#L27). If someone's username is too long and needs to be hashed, it would be possible to engineer a username which would allow you to take that user's order history from the cache. This can be avoided by using information that's not public in the cache key, like the user's ID.